src/components/behandlerdialog/meldingtilbehandler/VelgBehandler.tsx

```diff
@@ -1,53 +1,126 @@
-import React from "react";
-import { SkjemaelementFeilmelding } from "nav-frontend-skjema";
-import { Field } from "react-final-form";
+import React, { useEffect, useState } from "react";
 import { BehandlerDTO } from "@/data/behandler/BehandlerDTO";
 import AppSpinner from "@/components/AppSpinner";
-import BehandlerRadioGruppe, {
-  BehandlerRadioGruppeTexts,
-} from "@/components/behandler/BehandlerRadioGruppe";
 import { useBehandlereQuery } from "@/data/behandler/behandlereQueryHooks";
+import { Radio, RadioGroup } from "@navikt/ds-react";
+import {
+  FieldErrors,
+  UseFormRegister,
+  UseFormSetError,
+  UseFormTrigger,
+  UseFormWatch,
+} from "react-hook-form";
+import { MeldingTilBehandlerSkjemaValues } from "@/components/behandlerdialog/meldingtilbehandler/MeldingTilBehandlerSkjema";
+import { capitalizeWord } from "@/utils/stringUtils";
+import BehandlerSearch from "@/components/behandler/BehandlerSearch";
 
-const behandlerRadioGruppeTexts: BehandlerRadioGruppeTexts = {
+const texts = {
   behandlerLegend: "Velg behandler som skal motta meldingen",
   behandlersokTekst: "Søk etter behandler",
+  missingBehandler: "Vennligst velg behandler",
 };
 
 interface VelgBehandlerProps {
   selectedBehandler: BehandlerDTO | undefined;
   setSelectedBehandler: (behandler?: BehandlerDTO) => void;
+  register: UseFormRegister<MeldingTilBehandlerSkjemaValues>;
+  watch: UseFormWatch<MeldingTilBehandlerSkjemaValues>;
+  setError: UseFormSetError<MeldingTilBehandlerSkjemaValues>;
+  trigger: UseFormTrigger<MeldingTilBehandlerSkjemaValues>;
+  errors: FieldErrors<MeldingTilBehandlerSkjemaValues>;
 }
 
 export const VelgBehandler = ({
   selectedBehandler,
   setSelectedBehandler,
+  register,
+  watch,
+  trigger,
+  setError,
+  errors,
 }: VelgBehandlerProps) => {
   const { data: behandlere, isInitialLoading } = useBehandlereQuery();
-  const field = "behandlerRef";
+  const [showBehandlerSearch, setShowBehandlerSearch] =
+    useState<boolean>(false);
+
+  const behandlerOneliner = (behandler: BehandlerDTO): string => {
+    const name = [behandler.fornavn, behandler.mellomnavn, behandler.etternavn]
+      .filter(Boolean)
+      .join(" ");
+    const type = !!behandler.type ? `${capitalizeWord(behandler.type)}:` : "";
+    const typeAndName = `${type} ${name}`;
+    const office = !!behandler.kontor ? capitalizeWord(behandler.kontor) : "";
+    const phone = !!behandler.telefon ? `tlf ${behandler.telefon}` : "";
+
+    return [typeAndName, office, phone].filter(Boolean).join(", ");
+  };
+
+  const behandlerRefField = register("behandlerRef", {
+    required: true,
+    validate: (value) => {
+      if (value === "sokEtterBehandler" && selectedBehandler === undefined) {
+        setError("behandlerRef", { type: "custom" });
+        return false;
+      }
+      return true;
+    },
+  });
```

Comment:
Siden `sokEtterBehandler` radio button ikke er gyldig valg må vi ha dette custom validering opplegget. Radio button valget `sokEtterBehandler` er bare gyldig når man også har en verdi for `selectedBehandler`.

Reply:
Ser at jeg får valideringsfeil "Vennligst velg behandler" med en gang jeg velger Søk behandler (uten å ha trykket Sendt).

Reply:
Hmm, det er kanskje ikke ønsket oppførsel 🤔 Hva tenker du hvis man først har valgt en behandler i en radiobutton og deretter trykker "søk behandler" uten å velge faktisk behandler her? Jeg tenkte at man vil få valideringsfeil når man først har valgt noe som er gyldig og deretter velger noe som er ugyldig. Eller vil man alltid kun validere når man trykker submit?
Har vært borti at man validerer på `onBlur`, men tolker [dette](https://aksel.nav.no/god-praksis/artikler/skjemavalidering#:~:text=Noen%20folk%20utforsker,jobber%20i%20feltet.) som at man ikke skal gjøre det🤔

Reply:
Tror det er ønskelig at det ikke vises valideringsfeil før man trykker submit, ja. Også når man deretter gjør valg så skal valideringsfeilene forsvinne.

Reply:
Testet på nytt nå. Dersom jeg først trykker Send uten å fylle inn noe får jeg valideringsfeil, som er riktig. Men når jeg deretter velger Søk behandler, søker opp en og velger så forsvinner ikke valideringsfeilen slik den gjør for de andre feltene. Det gjelder altså kun når man velger en man søker opp og ikke når man velger en fra radioknappene. Sikkert ingen krise, men ikke helt etter boka :/

Reply:
Oi se der ja. Det bør fikses

Reply:
Skal være fikset nå
https://github.com/navikt/syfomodiaperson/assets/11747383/b50cfbb6-6bf7-4c46-961c-d02b4c6fe087

Reply:
Nice! Spotta dette på videoen:
<img width="455" alt="image" src="https://github.com/navikt/syfomodiaperson/assets/37441744/5d61cc41-f277-46f5-a2ef-e3f57811a773">

Vs den gamle:
<img width="525" alt="image" src="https://github.com/navikt/syfomodiaperson/assets/37441744/e409df58-1a8f-4e66-8aa2-a21bbf9e1ae7">

Bevisst valg at den kommer i større grad rett under radioknappen, med innrykk? Jeg har ikke noen formening om hva som er best altså, men mulig Andrea har 🤷🏼‍♂️

Reply:
Ja, tenkte egentlig det var logisk å koble den nærmere radio button'en, men veldig bra forslag å høre med Andrea 🙌 Feilmeldingen kommer også mellom disse elementene om man ikke legger denne search komponenten til som en del av `RadioGroup`'en